src/helpers.rs

```diff
@@ -883,3 +905,9 @@ impl std::fmt::Display for HexRange {
         write!(f, "[{:#x}..{:#x}]", self.0.start.bytes(), self.0.end().bytes())
     }
 }
+
+/// Helper function used inside the shims of foreign functions to check that the `target_os` is
+/// part of the UNIX family.
+pub fn target_os_is_unix(target_os: &str) -> bool {
```

Comment:
Elsewhere we are using `target.families.iter().any(|f| f == "unix")`. I think that probably makes more sense?

So, please change this to take a `&Target` and also adjust the existing checks to call this function instead.

Reply:
Actually, thinking about it more -- I like explicitly listing the unix OSes that we *do* support. So let's go with your approach.

But please change the existing code that checks `target.families` to call `target_os_is_unix` instead.

Reply:
Could you or @InfRandomness make a separate PR that does nothing but add `target_os_is_unix` and use it where appropriate? Then both the FreeBSD PR and this one can just adjust that function and don't need to change so many different places in the code.